src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-heal-node.ts

```diff
@@ -0,0 +1,277 @@
+                    ${this.hass.localize(
+                      "ui.panel.config.zwave_js.heal_node.healing_failed",
+                      {
+                        device: html`<em
+                          >${computeDeviceName(this.device, this.hass!)}</em
+                        >`,
+                      }
+                    )}
+                  </p>
+                  <p>
+                    ${this._error
+                      ? html` <em>${this._error}</em> `
+                      : `
+                  ${this.hass.localize(
+                    "ui.panel.config.zwave_js.heal_node.healing_failed_check_logs"
+                  )}
+                  `}
+                  </p>
+                </div>
+              </div>
+              <mwc-button slot="primaryAction" @click=${this.closeDialog}>
+                ${this.hass.localize("ui.panel.config.zwave_js.common.close")}
+              </mwc-button>
+            `
+          : ``}
+        ${this._status === "finished"
+          ? html`
+              <div class="flex-container">
+                <ha-svg-icon
+                  .path=${mdiCheckCircle}
+                  class="success"
+                ></ha-svg-icon>
+                <div class="status">
+                  <p>
+                    ${this.hass.localize(
+                      "ui.panel.config.zwave_js.heal_node.healing_complete",
+                      {
+                        device: html`<em
+                          >${computeDeviceName(this.device, this.hass!)}</em
+                        >`,
+                      }
+                    )}
+                  </p>
+                </div>
+              </div>
+              <mwc-button slot="primaryAction" @click=${this.closeDialog}>
+                ${this.hass.localize("ui.panel.config.zwave_js.common.close")}
+              </mwc-button>
+            `
+          : ``}
+        ${this._status === "network-healing"
+          ? html`
+              <div class="flex-container">
+                <ha-svg-icon
+                  .path=${mdiCloseCircle}
+                  class="failed"
+                ></ha-svg-icon>
+                <div class="status">
+                  <p>
+                    ${this.hass.localize(
+                      "ui.panel.config.zwave_js.heal_node.network_heal_in_progress"
+                    )}
+                  </p>
+                </div>
+              </div>
+              <mwc-button slot="primaryAction" @click=${this.closeDialog}>
+                ${this.hass.localize("ui.panel.config.zwave_js.common.close")}
+              </mwc-button>
+            `
+          : ``}
+      </ha-dialog>
+    `;
+  }
+
+  private async _fetchData(): Promise<void> {
+    if (!this.hass) {
+      return;
+    }
+    const network: ZWaveJSNetwork = await fetchNetworkStatus(
+      this.hass!,
+      this.entry_id!
+    );
+    if (network.controller.is_heal_network_active) {
+      this._status = "network-healing";
```

Comment:
Since we are not waiting for the network fetch call, technically a user could now start a new heal when one is still active.

Reply:
If you try to start a new heal while one is in progress the server will reject it, so our error handler will catch it.